source/adminguide/hosts.rst

```diff
@@ -845,3 +845,124 @@ Custom Script Execution Configuration
+
+#. The pre-flight scripts are executed on each host. If any of these scripts fail, then no action is performed unless the ‘force’ parameter is enabled.
+
+The pre-flight script may signal that no maintenance is needed on the host. In that case, the host is skipped from the rolling maintenance hosts iteration.
+
+Once pre-flight checks pass, then the management server iterates through each host in the selected scope and sends a command to execute each of the rest of the stages in order. The hosts in the selected scope are grouped by clusters, therefore all the hosts in a cluster are processed before processing the hosts of a different cluster. The management server iterates through hosts in each cluster on the selected scope and does the following:
+
+- Disables the cluster (if it has not been disabled previously) · The existence of the maintenance script on the host is checked (this check is performed only for the maintenance script, not for the rest of the stages)
+
+   - If the host does not contain a maintenance script, then the host is skipped and the iteration continues with the next host in the cluster.
+   
+-  Execute pre maintenance script (if any) before entering maintenance mode.
+
+   -  The pre-maintenance script may signal that no maintenance is needed on the host. In that case, the host is skipped and the iteration continues with the next host in the cluster.
+
+   -  In case the pre-maintenance script fails and the ‘forced’ parameter is not set, then the rolling maintenance process fails and an error is reported. If the ‘forced’ parameter is set, the host is skipped and the iteration continues with the next host in the cluster
+
+-  Capacity checks are recalculated, to verify that the host can enter maintenance mode.
+
+   .. note::
+      Before recalculating the capacity, the capacity is updated, similar to performing a listCapacity API execution, setting the ‘fetchLatest’ parameter to true
+
+· The host enters maintenance mode (throwing an error if the host does not enter maintenance after ‘kvm.rolling.maintenance.wait.maintenance.timeout’ seconds)
+
+· Execute maintenance script (if any) while the host is in maintenance.
+
+   - In case the maintenance script fails and the ‘forced’ parameter is not set, the rolling maintenance process fails, maintenance mode is cancelled and an error is reported. If the ‘forced’ parameter is set, the host is skipped and the iteration continues with the next host in the cluster
+
+· Cancel maintenance mode
+
+· Execute post maintenace script (if any) after cancelling maintenance mode.
+
+   - In case the post-maintenance script fails and the ‘forced’ parameter is not set, then the rolling maintenance process fails and an error is reported. If the ‘forced’ parameter is set, the host is skipped and the iteration continues with the next host in the cluster
```

Comment:
```suggestion
  - In case the post-maintenance script fails and the ‘forced’ parameter is not set, then the rolling maintenance process fails and an error is reported. If the ‘forced’ parameter is set, the host is skipped and the iteration continues with the next host in the cluster
```